libraries/c_sdk/standard/mqtt/src/iot_mqtt_subscription.c

```diff
@@ -350,10 +350,22 @@ IotMqttError_t _IotMqtt_AddSubscriptions( _mqttConnection_t * pMqttConnection,
                     connToContext[ contextIndex ].subscriptionArray[ index ].packetInfo.order = i;
                     connToContext[ contextIndex ].subscriptionArray[ index ].callback = pSubscriptionList[ i ].callback;
                     connToContext[ contextIndex ].subscriptionArray[ index ].topicFilterLength = pSubscriptionList[ i ].topicFilterLength;
-                    connToContext[ contextIndex ].subscriptionArray[ index ].pTopicFilter = IotMqtt_MallocMessage( pSubscriptionList[ i ].topicFilterLength );
-                    ( void ) memcpy( connToContext[ contextIndex ].subscriptionArray[ index ].pTopicFilter,
-                                     pSubscriptionList[ i ].pTopicFilter,
-                                     ( size_t ) ( pSubscriptionList[ i ].topicFilterLength ) );
+                    char * pTopicFilter = IotMqtt_MallocMessage( pSubscriptionList[ i ].topicFilterLength );
```

Comment:
Declare `pTopicFilter` at the beginning of the function.

Reply:
Addressed